lms/djangoapps/courseware/model_data.py

```diff
@@ -955,3 +922,23 @@ def last_modified(self, key):
 
     def __len__(self):
         return sum(len(cache) for cache in self.cache.values())
+
+
+# @contract(user_id=int, usage_key=UsageKey, score="number|None", max_score="number|None")
+def set_score(user_id, usage_key, score, max_score):
```

Comment:
I like this better, actually. It helps narrow down the `FieldDataCache` interface.